src/main/scala/mesosphere/marathon/api/v2/TasksResource.scala

```diff
@@ -107,9 +111,16 @@ class TasksResource @Inject() (
 
     groupedTasks.foreach {
       case (appId, tasks) =>
-        service.killTasks(appId, tasks, scale)
+        def findToKill(appTasks: Set[MarathonTask]) = tasks.toSet
+        if (scale) {
+          taskKiller.killAndScale(appId, findToKill, force = true)
+        }
+        else {
+          taskKiller.kill(appId, findToKill, force = true)
+        }
     }
 
+    // TODO: does anyone expect a response with all the deployment plans in case of scaling?
```

Comment:
Thanks for the TODO and indeed, it is a strange result. Also there should also only be one deployment.